sdk/samples/iot/hub/src/sample_pnp_thermostat_component.c

```diff
@@ -0,0 +1,368 @@
+  {
+    printf(
+        "PnP property=%.*s is not supported on thermostat component\r\n",
+        az_span_size(property_name),
+        az_span_ptr(property_name));
+  }
+
+  double parsed_value = 0;
+  if (az_failed(az_json_token_get_double(property_value, &parsed_value)))
+  {
+    result = pnp_helper_create_reported_property_with_status(
+        mqtt_message->payload_span,
+        component_name,
+        property_name,
+        append_double,
+        (void*)&parsed_value,
+        401,
+        version,
+        temp_response_description_failed,
+        &mqtt_message->out_payload_span);
+  }
+  else
+  {
+    handle->current_temperature = parsed_value;
+    if (handle->current_temperature > handle->max_temperature)
+    {
+      handle->max_temperature = handle->current_temperature;
+      handle->send_max_temp_property = true;
+    }
+
+    if (handle->current_temperature < handle->min_temperature)
+    {
+      handle->min_temperature = handle->current_temperature;
+    }
+
+    /* Increment the avg count, add the new temp to the total, and calculate the new avg */
+    handle->device_temperature_avg_count++;
+    handle->device_temperature_avg_total += handle->current_temperature;
+    handle->avg_temperature
+        = handle->device_temperature_avg_total / handle->device_temperature_avg_count;
+
+    if ((result = pnp_helper_create_reported_property_with_status(
+             mqtt_message->payload_span,
+             component_name,
+             property_name,
+             append_double,
+             (void*)&parsed_value,
+             200,
+             version,
+             temp_response_description_success,
+             &mqtt_message->out_payload_span))
+        != AZ_OK)
+    {
+      printf("Error to get reported property payload with status: error code = 0x%08x", result);
+    }
+  }
+
+  if ((result = az_iot_hub_client_twin_patch_get_publish_topic(
+           client, get_request_id(), mqtt_message->topic, mqtt_message->topic_length, NULL))
+      != AZ_OK)
+  {
+    printf("Error to get reported property topic with status: error code = 0x%08x", result);
+  }
+
+  return result;
+}
+
+az_result sample_pnp_thermostat_process_command(
+    az_iot_hub_client* client,
+    sample_pnp_thermostat_component* handle,
+    az_iot_hub_client_method_request* command_request,
+    az_span component_name,
+    az_span command_name,
+    az_span command_payload,
+    sample_pnp_mqtt_message* mqtt_message)
+{
+  az_result result;
+
+  if (az_span_is_content_equal(handle->component_name, component_name)
+      && az_span_is_content_equal(report_command_name_span, command_name))
+  {
+    // Invoke command
+    uint16_t return_code;
+    az_result response = invoke_getMaxMinReport(
+        handle, command_payload, mqtt_message->payload_span, &mqtt_message->out_payload_span);
+    if (response != AZ_OK)
+    {
+      return_code = 400;
+    }
+    else
+    {
+      return_code = 200;
+    }
+
+    if (az_failed(
```

Comment:
>az_failed [](start = 8, length = 9)

RETURN_IF_FAILED

Reply:
the return code would not be printed if we did that